### DISCUSSION TITLE
[Discussion] XAML Syntax for MAUI?

### DISCUSSION THREAD

**Discussion by dfarrNTST** [dotnet/maui#198] [General]
Not sure if this is even a item that can be updated at this point in development, but I would love to get everyone's opinion on the "featured" XAML syntax that should be in MAUI.

There are differences in XAML from each UI implementation over the years, I am partial to WPF and have not enjoyed the nuances of XF's version of XAML.  So far it sounds as if the XF version of XAML is going to be used in MAUI but I'm hoping there is time and opportunity for this to be discussed and, if preferred by the majority of developers, changed.

What are the benefits of each XAML syntax?  How difficult would it be for MS to use WPF's version of XAML as the syntax for it's implementation with MAUI?

Some of the XF Syntax has always been confusing (such as Height vs HeightRequest, what does MinimumHeightRequest actually do (why doesn't it actually make the Height of something have a minimum height?, and various other syntax decisions that went away from WPF that don't make any sense.)  I do know what these do after using it for a long time, but when we were starting out with Xamarin years ago it made no sense and was confusing.  Missing lifecycle events such as Loading/Loaded/etc.  Full Gesture support!  This seems like the perfect time to fix these items and utilize a better "XAML Standard" type approach that is far less confusing.

My recommendation would be to fix these XAML syntax issues for MAUI for clarity and the longevity of the product.  We are going to be investigating Flutter and some other platforms over the next 6-12 months and it would certainly help if developing with MAUI was easier to understand and implement for beginners and intermediate developers.

Is it possible to have current XF XAML syntax a supported syntax, if enabled, for backwards compatibility but a better, cleaner, and clearer "XAML Standard"-like syntax as the default?

What are everyone's thoughts and ideas on this subject?

**Comment by schmidge** [comment]
I agree with your recommendation.

**Comment by lylerolleman** [comment]
I don't think syntax is the right word to use here. Height vs HeightRequest has nothing to do with syntax, rather, they are what properties are exposed on the controls

XAML is a clearly defined syntactic standard, and definitely will not be changing

As for what properties each control has, or what features/events are available on each control/layout, that can probably be debated and possibly even improved. It is however a rather huge topic so I'd recommend pairing down discussions to specific controls (a wholesale "just use WPF" certainly isn't going to be in the cards)

**Comment by ccarlo88** [comment]
@dfarrNTST don't waste your time. This issue has already been discussed here: https://github.com/dotnet/maui/issues/43
#BringBallmerBack
